php/libraries/VisitControler.class.inc

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * This file contains the VisitControler class.
+ *
+ * PHP Version 7
+ *
+ * @category Controler
+ * @package  Visits
+ * @author   Mélanie Legault <melanie.legault2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS;
+
+/**
+ * Base class that contains method to access the visit of projects
+ *
+ * @category Controler
+ * @package  Visits
+ * @author   Mélanie Legault <melanie.legault2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class VisitControler
+{
+    /**
+     * The database connection to use
+     *
+     * @var \Database
+     */
+    protected $database;
+
+    /**
+     * The constructor
+     *
+     * @param \Database $database The database connection to use
+     */
+    public function __construct(\Database $database)
+    {
+        if (!$database->isConnected()) {
+            throw new \RuntimeException("VisitControler - Database not connected");
+        }
+        $this->database = $database;
+    }
+
+    /**
+     * Retruns a list of configured Visits
+     *
+     * @return array The list of visits
+     */
+    public function getAllVisits(): array
+    {
+        return array_map(
```

Comment:
This whole map seems like it could be avoided by just using pselectCol instead of pselect, since there's only one field.

And is there a reason for this to return an array of visit (strings) rather than the Visit (object) type you defined above?

Reply:
To make an easy replacement for the Utility::getVisitList()